autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -1193,6 +1338,182 @@ async def get_pending_submissions(
+        # Build the where clause for StoreListing
+        where_dict = {}
+        where_dict["isDeleted"] = False
+
+        sanitized_query = sanitize_query(search_query)
+        if sanitized_query:
+            # Find users with matching email
+            matching_users = await prisma.models.User.prisma().find_many(
+                where={"email": {"contains": sanitized_query, "mode": "insensitive"}},
+            )
+
+            user_ids = [user.id for user in matching_users]
+
+            # Set up OR conditions
+            where_dict["OR"] = [
+                {"slug": {"contains": sanitized_query, "mode": "insensitive"}},
+                {
+                    "Versions": {
+                        "some": {
+                            "name": {"contains": sanitized_query, "mode": "insensitive"}
+                        }
+                    }
+                },
+                {
+                    "Versions": {
+                        "some": {
+                            "description": {
+                                "contains": sanitized_query,
+                                "mode": "insensitive",
+                            }
+                        }
+                    }
+                },
+                {
+                    "Versions": {
+                        "some": {
+                            "subHeading": {
+                                "contains": sanitized_query,
+                                "mode": "insensitive",
+                            }
+                        }
+                    }
+                },
+            ]
+
+            # Add user_id condition if any users matched
+            if user_ids:
+                where_dict["OR"].append({"owningUserId": {"in": user_ids}})
+
+        # Calculate pagination
+        skip = (page - 1) * page_size
+
+        # Create proper Prisma types for the query
+        where = prisma.types.StoreListingWhereInput(**where_dict)
+        include = prisma.types.StoreListingInclude(Versions=True, OwningUser=True)
+
+        # Query listings with their versions
+        listings = await prisma.models.StoreListing.prisma().find_many(
+            where=where,
+            skip=skip,
+            take=page_size,
+            include=include,
+            order=[{"createdAt": "desc"}],
+        )
+
+        # Get total count for pagination
+        total = await prisma.models.StoreListing.prisma().count(where=where)
+        total_pages = (total + page_size - 1) // page_size
+
+        # Convert to response models
+        listings_with_versions = []
+        for listing in listings:
+            versions = []
+            # Filter versions by status if specified and ensure versions is not None
+            if listing.Versions is not None:
+                filtered_versions = [
+                    v
+                    for v in listing.Versions
+                    if status is None or v.submissionStatus == status
+                ]
```

Comment:
If we filter it on the db side, we dont' need to fetch all listing in the world and filtering it here.